Init PR: create rapruc module, update era5 module, update run module, add sounderpy to .toml

kylejgillett opened the pull request:
Hopefully I took care of everything needed. There may be some more testing needed on your end just to make sure though.

wxmann commented:
Have you tried running it locally and generating some profiles?

That said, LGTM! The only thing I would modify is putting the entire `.idea` folder in the root `.gitignore`.

wxmann commented:
Also forgot to ask -- can you add the dependency to `pyproject.toml` via a `poetry add sounderpy`?

kylejgillett commented:
- Added ``.ideas`` to the ``.gitignore``
- installed poetry and successfully generated two new .json files
- added ``sounderpy`` to the ``.toml``
- updated ``models.Profile()`` & ``run._to_dict()`` to include `"source"` (str) as an argument. For ERA5 data, the "source" is `"ERA5"`, for RAP data it is `"RAP"`, for RUC data it is `"RUC"`.

wxmann commented:
> * Added `.ideas` to the `.gitignore`
> * installed poetry and successfully generated two new .json files
> * added `sounderpy` to the `.toml`
> * updated `models.Profile()` & `run._to_dict()` to include `"source"` (str) as an argument. For ERA5 data, the "source" is `"ERA5"`, for RAP data it is `"RAP"`, for RUC data it is `"RUC"`.

Thank you 🙏 

However I see the `.idea` files still in the changelist, could you remove those files from that? Ideally the changes should just include the code changes and not IDE config files (since different people might use different IDE's)

Was able to visualize a sounding from [b18e3bb0e932340e8a17bf39c5fd935c.json](https://github.com/chase-archive/weather-case-search/pull/4/files#diff-0f17fa7728d933a85d89328985b44ede4b066e798692cd429892c25579e0ba06) but not from [9c52cec4146b954219cda3a177fa5758.json](https://github.com/chase-archive/weather-case-search/pull/4/files#diff-6b5ded20976cdf64e49a6c8859dcd25a4da6455ffb4f9d739e61468f261a9400) - which case is that? Trying to find it via the API does not seem to work for me (may just be on my end though)

https://urchin-app-tpil4.ondigitalocean.app/docs#/cases/get_case_by_id_cases__case_id__get

<img width="733" alt="image" src="https://github.com/user-attachments/assets/e960d38c-e74f-431a-aef0-0cb947615e15" />

kylejgillett commented:
I can see both test cases locally and pulled up here in this PR. 

Case [9c52cec4146b954219cda3a177fa5758.json](https://github.com/chase-archive/weather-case-search/pull/4/files#diff-6b5ded20976cdf64e49a6c8859dcd25a4da6455ffb4f9d739e61468f261a9400) is ``"lat": 34.16, "lon": 261.9, "timestamp": "2024-11-03T06:00:00"``. The file exists, but wont show up for me either using the API.

Pushed a new commit that wipes the cache for `.idea` and updates the `.gitignore` such that `.idea` is no long included in the change list. 

For some reason GitHub didn't like my commit message lol

wxmann commented:
> I can see both test cases locally and pulled up here in this PR.
> 
> Case [9c52cec4146b954219cda3a177fa5758.json](https://github.com/chase-archive/weather-case-search/pull/4/files#diff-6b5ded20976cdf64e49a6c8859dcd25a4da6455ffb4f9d739e61468f261a9400) is `"lat": 34.16, "lon": 261.9, "timestamp": "2024-11-03T06:00:00"`. The file exists, but wont show up for me either using the API.
> 
> Pushed a new commit that wipes the cache for `.idea` and updates the `.gitignore` such that `.idea` is no long included in the change list.
> 
> For some reason GitHub didn't like my commit message lol

Cool, sounds good. I'm not sure what's going on with the extra case but nbd I see it in the master spreadsheet.

Approved, feel free to merge whenever

BTW in the future feel free to just branch off the original repo instead of forking it, since we're both core contributors to the project.
